src/supa/job/dataplane.py

```diff
@@ -129,11 +152,283 @@ def trigger(self) -> DateTrigger:
         Returns:
             DateTrigger set to start_time of reservation.
         """
+        return DateTrigger(run_date=None)  # Run immediately
+
+
+class DeactivateJob(Job):
+    """Handle data plane deactivation requests."""
+
+    connection_id: UUID
+    log: BoundLogger
+
+    def __init__(self, connection_id: UUID):
+        """Initialize the DeactivateJob.
+
+        Args:
+           connection_id: The connection_id of the reservation deactivation request
+        """
+        self.log = logger.bind(job=self.__class__.__name__, connection_id=str(connection_id))
+        self.connection_id = connection_id
+
+    def __call__(self) -> None:
+        """Deactivate data plane request.
+
+        Deactivate the data plane and
+        send a data plane status notitication to the NSA/AG.
+        If the data plane state machine is not in the correct state for a Deactivate
+        a NSI exception is returned leaving the state machine unchanged.
+        """
+        self.log.info("Deactivating data plane")
+
+        from supa.db.session import db_session
+
+        response: Union[DataPlaneStateChangeRequest, ErrorRequest]
+        with db_session() as session:
+            reservation = (
+                session.query(Reservation).filter(Reservation.connection_id == self.connection_id).one_or_none()
+            )
+            dpsm = DataPlaneStateMachine(reservation, state_field="data_plane_state")
+            # lsm = LifecycleStateMachine(reservation, state_field="lifecycle_state")
+            # # when past end time register a lifecycle end time event
+            # if reservation.end_time <= current_timestamp():
+            #     lsm.endtime_event()
+            #     dpsm.deactivate_request()
+            try:
+                #
+                # TODO:  Call the Network Resource Manager to deactivate the data plane.
```

Comment:
On purpose?

Reply:
The interface to the NRM needs to be implemented first before this part of the code can be finished.